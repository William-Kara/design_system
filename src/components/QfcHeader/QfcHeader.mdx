import { ArgTypes, Meta } from '@storybook/blocks';
import QfcHeader from '@/components/QfcHeader/QfcHeader.vue';

<Meta title="Modules/Headers/Header" />

# Definition

QfcHeader is placed at the top of the page. It contains a logo and includes the QfcNav component.

## Props

The following props are available for the QfcHeader component:

<ArgTypes of={QfcHeader} />

# Usage

To use QfcHeader, style needs to be imported separately.

```sass
@import '@website/design-system-vuejs/dist/styles/bulma/components/navbar.scss';
@import '@website/design-system-vuejs/dist/styles/components/QfcHeader.scss';
```

```html
<template>
  <QfcHeader :navItems="navItems" :endItems="endItems"> </QfcHeader>
</template>
```
