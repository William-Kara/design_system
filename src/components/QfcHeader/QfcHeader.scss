.navbar {
  .navbar-menu {
    &__header {
      display: none;
      padding-top: 2rem;
      margin-right: 2rem;
      font-size: 2.8rem;
      font-weight: 700;
    }
    &__title {
      margin-bottom: 4rem;
    }
  }
  padding-top: 1rem;
  .navbar-brand {
    padding-left: 6.525rem;
    .navbar-item {
      align-items: center;
      height: 100%;
      img {
        margin-bottom: 0.5rem;
      }
    }
  }
  .navbar-end {
    padding-right: 7.5rem;
  }
}
@include touch {
  .navbar {
    .navbar-menu {
      padding-left: 2rem;
      &.is-active {
        .navbar-menu__header {
          display: block;
        }
      }
    }
    padding-top: 1rem;
    .navbar-brand {
      padding-left: 1rem;
      .navbar-item {
        img {
          margin-top: 0.5rem;
        }
      }
    }
    .navbar-end {
      padding-right: 0;
      font-size: 1.25rem;
      .navbar-item {
        margin-top: 3rem;
      }
    }
    .navbar-center {
      padding-top: 1.5rem;
      font-size: 1.25rem;
    }
    .navbar-burger {
      margin-right: 2rem;
      &.is-active {
        span {
          &:nth-child(1) {
            transform: translateY(0.5rem) rotate(45deg);
          }
          &:nth-child(3) {
            transform: translateY(-0.5rem) rotate(-45deg);
          }
        }
      }

      span {
        width: 1.75rem;
        &:nth-child(1) {
          top: calc(50% - 0.625rem);
        }
        &:nth-child(2) {
          top: calc(50% - 0.125rem);
        }
        &:nth-child(3) {
          top: calc(50% + 0.375rem);
        }
      }
    }
  }
}
