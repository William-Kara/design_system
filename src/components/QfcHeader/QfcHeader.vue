<script lang="ts" setup>
import { ref } from 'vue';
import QfcNav from './../QfcNav';
import QfcNavItem from './../QfcNav/QfcNavItem.vue';
import QfcSeparator from './../QfcSeparator';
interface NavItems {
  label: string;
  to: string;
}
interface Props {
  centerNavItems: NavItems[];
  rightNavItems: NavItems[];
}
withDefaults(defineProps<Props>(), {});

const isActive = ref(false);
const toggleActive = () => {
  isActive.value = !isActive.value;
};
</script>
<template>
  <div class="navbar">
    <div class="navbar-brand">
      <a class="navbar-item">
        <img src="/logo.svg" width="168" height="18" alt="logo" />
      </a>
      <a
        role="button"
        class="navbar-burger"
        aria-label="menu"
        :aria-expanded="isActive"
        :class="{ 'is-active': isActive }"
        @click="toggleActive()"
      >
        <span aria-hidden="true"></span>
        <span aria-hidden="true"></span>
        <span aria-hidden="true"></span>
      </a>
    </div>
    <div :class="{ 'is-active': isActive }" class="navbar-menu">
      <div class="navbar-menu__header navbar-item">
        <div class="navbar-menu__title">Menu.</div>
        <QfcSeparator class="navbar-menu__separator" />
      </div>
      <QfcNav :navItems="centerNavItems" />
      <div class="navbar-end">
        <QfcNavItem
          v-for="({ label, to, isExternal }, index) in rightNavItems"
          :key="index"
          :label="label"
          :to="to"
          :isExternal="isExternal"
        />
      </div>
    </div>
  </div>
</template>

<style lang="scss" scoped src="./QfcHeader.scss" />
