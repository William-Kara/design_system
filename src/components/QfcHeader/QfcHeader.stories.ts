import QfcHeader from './QfcHeader.vue';
import type { Meta, StoryObj } from '@storybook/vue3';

const meta: Meta<typeof QfcHeader> = {
  component: QfcHeader,
  title: 'Modules/Headers/Header',
};

export default meta;
type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  args: {
    centerNavItems: [
      { label: 'Title 1', to: '/' },
      { label: 'Title 2', to: '/' },
      { label: 'Title 3', to: '/' },
    ],
    rightNavItems: [{ label: 'Link', to: 'https://test', isExternal: true }],
  },
};
