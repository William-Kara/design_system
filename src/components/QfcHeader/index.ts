import QfcHeader from './QfcHeader.vue';

export { QfcHeader };
export default QfcHeader;
