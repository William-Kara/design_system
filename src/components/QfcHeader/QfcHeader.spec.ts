import { it, describe, beforeEach, expect } from 'vitest';

import { shallowMount } from '@vue/test-utils';
import QfcHeader from './QfcHeader.vue';

let wrapper: any = null;

const propsValues = {
  centerNavItems: [
    { label: 'Title 1', to: '' },
    { label: 'Title 2', to: '' },
    { label: 'Title 3', to: '' },
    { label: 'Title 4', to: '' },
  ],
  rightNavItems: [{ label: 'Title 5', to: 'https://test', isExternal: true }],
};

beforeEach(() => {
  wrapper = shallowMount(QfcHeader, {
    propsData: propsValues,
  });
});

describe('QfcHeader Component', () => {
  it('should mount properly', () => {
    expect(wrapper).toBeTruthy();
  });

  it('should have correct props', () => {
    expect(wrapper.props()).toEqual(propsValues);
  });

  it('should render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
