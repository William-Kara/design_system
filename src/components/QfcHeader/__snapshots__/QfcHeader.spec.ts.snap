// Vitest Snapshot v1

exports[`QfcHeader Component > should render correctly 1`] = `
"<div data-v-4ee6cf6a=\\"\\" class=\\"navbar\\">
  <div data-v-4ee6cf6a=\\"\\" class=\\"navbar-brand\\"><a data-v-4ee6cf6a=\\"\\" class=\\"navbar-item\\"><img data-v-4ee6cf6a=\\"\\" src=\\"/logo.svg\\" width=\\"168\\" height=\\"18\\" alt=\\"logo\\"></a><a data-v-4ee6cf6a=\\"\\" role=\\"button\\" class=\\"navbar-burger\\" aria-label=\\"menu\\" aria-expanded=\\"false\\"><span data-v-4ee6cf6a=\\"\\" aria-hidden=\\"true\\"></span><span data-v-4ee6cf6a=\\"\\" aria-hidden=\\"true\\"></span><span data-v-4ee6cf6a=\\"\\" aria-hidden=\\"true\\"></span></a></div>
  <div data-v-4ee6cf6a=\\"\\" class=\\"navbar-menu\\">
    <div data-v-4ee6cf6a=\\"\\" class=\\"navbar-menu__header navbar-item\\">
      <div data-v-4ee6cf6a=\\"\\" class=\\"navbar-menu__title\\">Menu.</div>
      <qfc-separator-stub data-v-4ee6cf6a=\\"\\" thickness=\\"1px\\" class=\\"navbar-menu__separator\\"></qfc-separator-stub>
    </div>
    <qfc-nav-stub data-v-4ee6cf6a=\\"\\" navitems=\\"[object Object],[object Object],[object Object],[object Object]\\"></qfc-nav-stub>
    <div data-v-4ee6cf6a=\\"\\" class=\\"navbar-end\\">
      <qfc-nav-item-stub data-v-4ee6cf6a=\\"\\" label=\\"Title 5\\" to=\\"https://test\\" isexternal=\\"true\\"></qfc-nav-item-stub>
    </div>
  </div>
</div>"
`;
